src/util.ts

```diff
@@ -518,3 +533,57 @@ export async function promptToInstallRPackage(name: string, section: string, cwd
             }
         });
 }
+
+/**
+ * Create temporary directory. Will avoid name clashes. Caller must delete directory after use.
+ * 
+ * @param root Parent folder.
+ * @param hidden If set to true, directory will be prefixed with a '.' (ignored on windows).
+ * @returns Path to the temporary directory.
+ */
+export function createTempDir(root: string, hidden?: boolean): string {
+    const hidePrefix = (!hidden || process.platform === 'win32') ? '' : '.';
+    let tempDir: string;
+    while (fs.existsSync(tempDir = path.join(root, `${hidePrefix}___temp_${randomBytes(8).toString('hex')}`))) { /* Name clash */ }
+    fs.mkdirSync(tempDir);
+    return tempDir;
+}
+
+/**
+ * Utility function for converting 'unknown' types to errors.
+ * 
+ * Usage:
+ * 
+ * ```ts
+ * try { ... } 
+ * catch (e) { 
+ *  const err: Error = catchAsError(e); 
+ * }
+ * ```
+ * @param err 
+ * @param fallbackMessage 
+ * @returns 
+ */
+export function catchAsError(err: unknown, fallbackMessage?: string): Error {
+    return (err instanceof Error) ? err : Error(fallbackMessage ?? 'Unknown error');
+}
+
+
+
+export function asViewColumn(s: string | undefined | vscode.ViewColumn): vscode.ViewColumn | undefined;
+export function asViewColumn(s: string | undefined | vscode.ViewColumn, fallback: vscode.ViewColumn): vscode.ViewColumn;
+export function asViewColumn(s: string | undefined | vscode.ViewColumn, fallback?: vscode.ViewColumn): vscode.ViewColumn | undefined {
+    if(!s){
+        return fallback;
+    }
+    if(typeof s !== 'string'){
+        // s is already ViewColumn:
+        return s;
+    }
+    if(Object.keys(vscode.ViewColumn).includes(s)){
+        const sKey = s as keyof typeof vscode.ViewColumn;
+        return(vscode.ViewColumn[sKey]);
```

Comment:
This is not entirely correct. `Object.keys(SomeEnumType)` will also include it's stringified values.
([TS Playground](https://www.typescriptlang.org/play?#code/FAUwdgrgtgBAagSxAdwMIHsA20wwN7AxHFED0AVOYSSeTAIIzkDOAnlAEZYIDGdIAEwQAXdACcYPLDhhiQABznNwwhGADmMYQAsQkiGLlhhmVjACGPVQDc9U7FDAA6GABVtCZjGvnsIajREdDzmuBx6EMoCMMi6uOjy4GqagiLizAA0MBwQwlq6TEpYtgJ0eAACmGoA1m4gAB7CAKJComJO1kho0o7eXRgOYAC+ALQ+fgGBdOgAZjCpbV7ICJiYFpjI5qxe4TAABgDyYCB7WXuuyOin++5yJxlOjzDi+xBgAiAzaoJ72bkwx1sEj29CsCFseyckxo5FI0OIoJsegAvDARgBGDLwsiUbFBBhMNicbh8eatF72GRyRQgZTGZL5Ow9XCiRkwZgID7POY6PSWJHPY4udyeby+CD+QLEYKhbIRKIxOLPRJgBkLdJZHJ5XmFWnFQRlSo1OqNFppdqdFADGSW7qDUbjCV4pjcsnmpYrNa+TbbOX7I73G6Xa7nbR3U6PFwvPZvD5fY4CX5agEgIH7ABCtM5JyhUqCcLzmY5XNRIwATFi8xQqHmXe49F8xMw8uqJJTHLm87C8QGYKjMXjq866PX2SApO83W1JMzO1Lu3mLug+zAK4PcbWRwUdAgxNFWzPBnOpgWpbcQCiYABmStSoebtwFGboAw6KcU2fD0+BABiL4kqIACy3oE95do+DYIDMb4Hu2zhfniP7gpeACsIE0GB84QeyCCNNo75tp+m7fjQADKuErgAbOhJCYVM2HKLYxj4bBRFdiRJCkam4ArgA7DRxB0TC2EgAg6jaDB5KEUeCF5k0YkSSuAAcAk4jW4GjqqzEEYeODHjCHHEAAct8K4AJyTEMwDABOzBYCATiYOg6gABQHBwABW47CE41QgNsLmIFazIAJQhdZpCkG4y6YCA5hiLgUDiHyXD-DqmChOoEDmOoIBZDwVQ8LU5hcLYMBqDAABETT1OYUDyLFzCVc8EiVQA6to5jCAA5F4RkoJVUKRTABy8mIyzKFkuV5M28XCII2RmHISWdBojLKDOUBQCoXihNEOqXGImDRKKrD-jADVbOoYgvu8nZAA))

I am actually not sure there is a way to check this properly.

Reply:
`Object.keys(ViewColumn).filter(x => isNaN(parseInt(x)))`

Feels a bit hacky but would work.

Reply:
fixed now